### PR TITLE
Rename `legible_unique_name` param to `force_readable_name`

### PR DESCRIPTION
**Pull request by Mickeon**
With the introduction of **Scene Unique Nodes**, `is_unique_in_owner`, "_Unique Name in Scene_" and other descriptions related to the feature, the second parameter of **Node**.`add_child()` and `add_simbling()` could be misunderstood to be related, at first glance. It is not.

This PR attempts to address that.

~This PR was previously renaming to `legible_name`~

Can be cherrypicked for 3.x.

### CODE REVIEW

**Review by fire-forge — APPROVED** [review]

**Review by aaronfranke — APPROVED** [review]

**Comment by akien-mga** [comment]
Long bikeshedding, agreed on using `force_readable_name` (should also be renamed in `_validate_property`).

The docs need to be improved to clarify the actual disambiguation logic that runs based on this property.

**Comment by Mickeon** [comment]
I can't find the argument name in `_validate_property`, are you sure?

**Comment by Mickeon** [comment]
I spent hours on a new description after many iterations and testing, I hope it was worth it.

**Comment by YuriSizov** [comment]
> I spent hours on a new description after many iterations and testing, I hope it was worth it.

I think it's a bit too wordy and would split it into several sentences, following "If true does this. In this case, this happens. In this other case, that happens. Performance is slow, so use with caution"

**Comment by Mickeon** [comment]
Attempted to follow the advice and reduce some verbosity.

File: doc/classes/Node.xml [hunk 1]
```diff
@@ -121,25 +121,25 @@
...
 				If [param internal] is different than [constant INTERNAL_MODE_DISABLED], the child will be added as internal node. Such nodes are ignored by methods like [method get_children], unless their parameter [code]include_internal[/code] is [code]true[/code].The intended usage is to hide the internal nodes from the user, so the user won't accidentally delete or modify them. Used by some GUI nodes, e.g. [ColorPicker]. See [enum InternalMode] for available modes.
 				[b]Note:[/b] If the child node already has a parent, the function will fail. Use [method remove_child] first to remove the node from its current parent. For example:
 				[codeblocks]
 				[gdscript]
 				var child_node = get_child(0)
 				if child_node.get_parent():
-				    child_node.get_parent().remove_child(child_node)
+				child_node.get_parent().remove_child(child_node)
```
Comment (akien-mga) [inline doc/classes/Node.xml:135]: You mistakenly removed the indentation here and in the C# code. In code blocks, code should be indented with 4 spaces after the normal tab-based XML indentation.

**Review by akien-mga — APPROVED** [review]
Looks good, just a style issue in the docs to fix.

**Comment by akien-mga** [comment]
Thanks!

Merged manually as f577bae76f85575c3bebebbc90c7a0120a66a8db (should have been auto-closed and shown as merged but I wrongly rebased locally after doing a local merge...).
